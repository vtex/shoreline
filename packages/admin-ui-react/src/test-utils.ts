import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
