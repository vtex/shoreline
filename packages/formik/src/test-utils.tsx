import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
