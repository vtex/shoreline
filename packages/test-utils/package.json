{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
