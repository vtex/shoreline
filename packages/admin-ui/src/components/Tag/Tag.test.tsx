import React from 'react'
import { render, axe } from '../../test-utils'
import { IconPlus } from '@vtex/phosphor-icons'

import { Tag } from './index'

describe('Tag tests', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(
      <>
        <Tag label="Rio de Janeiro" palette="cyan" />
        <Tag label="Rio de Janeiro" palette="purple" />
        <Tag label="Rio de Janeiro" palette="green" />
        <Tag label="Rio de Janeiro" palette="red" />
        <Tag label="Rio de Janeiro" palette="purple" />
        <Tag label="Rio de Janeiro" palette="orange" />
        <Tag label="Rio de Janeiro" palette="teal" />
        <Tag label="Rio de Janeiro" palette="gray" />
        <Tag label="tag-1" />
        <Tag label="tag-2" size="regular" />
        <Tag label="tag-3" size="small" />
        <Tag label="tag-4" handleDelete={() => {}} />
        <Tag label="tag-5" icon={<IconPlus />} handleDelete={() => {}} />
      </>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
