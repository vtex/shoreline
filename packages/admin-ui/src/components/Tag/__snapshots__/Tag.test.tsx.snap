// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(190, 58%, 90%, 1);
  color: hsla(187, 77%, 15%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-0 button {
  height: 20px;
  width: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(265, 35%, 93%, 1);
  color: hsla(268, 38%, 28%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-0 button {
  height: 20px;
  width: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(123, 39%, 91%, 1);
  color: hsla(123, 41%, 19%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-0 button {
  height: 20px;
  width: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(0, 60%, 93%, 1);
  color: hsla(0, 58%, 30%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-0 button {
  height: 20px;
  width: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(265, 35%, 93%, 1);
  color: hsla(268, 38%, 28%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-0 button {
  height: 20px;
  width: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(30, 100%, 92%, 1);
  color: hsla(29, 100%, 18%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-0 button {
  height: 20px;
  width: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(166, 52%, 87%, 1);
  color: hsla(171, 100%, 14%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-0 button {
  height: 20px;
  width: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 23%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-0 button {
  height: 20px;
  width: 20px;
}

<div
    class="emotion-0"
  >
    Rio de Janeiro
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(206, 61%, 92%, 1);
  color: hsla(207, 64%, 22%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-0 button {
  height: 20px;
  width: 20px;
}

<div
    class="emotion-0"
  >
    tag-1
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(206, 61%, 92%, 1);
  color: hsla(207, 64%, 22%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-0 button {
  height: 20px;
  width: 20px;
}

<div
    class="emotion-0"
  >
    tag-2
  </div>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(206, 61%, 92%, 1);
  color: hsla(207, 64%, 22%, 1);
  height: 26px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.emotion-0 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-0 svg {
  width: 16px;
  height: 16px;
  min-width: 16px;
  min-height: 16px;
}

.emotion-0 button {
  height: 16px;
  width: 16px;
}

<div
    class="emotion-0"
  >
    tag-3
  </div>
  .emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(206, 61%, 92%, 1);
  color: hsla(207, 64%, 22%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-4 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-4 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-4 button {
  height: 20px;
  width: 20px;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 32px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  color: #000000;
  background: transparent;
  margin-left: 0.25rem;
}

.emotion-3:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-3:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-3:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-3 svg {
  margin: 0.25rem;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-0 {
  width: 1rem;
  height: 1rem;
  min-height: 1rem;
  min-width: 1rem;
}

<div
    class="emotion-4"
  >
    tag-4
    <button
      aria-label="tag-4"
      class="emotion-3"
      type="button"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-1"
        >
          <svg
            class="emotion-0"
            fill="currentColor"
            height="1rem"
            viewBox="0 0 256 256"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16rem"
              width="16rem"
            />
            <line
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
              x1="200"
              x2="56"
              y1="56"
              y2="200"
            />
            <line
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
              x1="200"
              x2="56"
              y1="200"
              y2="56"
            />
          </svg>
           
        </div>
      </div>
    </button>
  </div>
  .emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100px;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  border: none;
  background: hsla(206, 61%, 92%, 1);
  color: hsla(207, 64%, 22%, 1);
  height: 40px;
  padding-left: 1rem;
  padding-right: 1rem;
}

.emotion-5 >svg:nth-of-type(1) {
  margin-right: 0.25rem;
}

.emotion-5 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
}

.emotion-5 button {
  height: 20px;
  width: 20px;
}

.emotion-4 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 32px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  color: #000000;
  background: transparent;
  margin-left: 0.25rem;
}

.emotion-4:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-4:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-4:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-4:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-4:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-4 svg {
  margin: 0.25rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-0 {
  width: 1rem;
  height: 1rem;
  min-height: 1rem;
  min-width: 1rem;
}

<div
    class="emotion-5"
  >
    <svg
      class="emotion-0"
      fill="currentColor"
      height="1rem"
      viewBox="0 0 256 256"
      width="1rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="16rem"
        width="16rem"
      />
      <line
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="18"
        x1="40"
        x2="216"
        y1="128"
        y2="128"
      />
      <line
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="18"
        x1="128"
        x2="128"
        y1="40"
        y2="216"
      />
    </svg>
    tag-5
    <button
      aria-label="tag-5"
      class="emotion-4"
      type="button"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-2"
        >
          <svg
            class="emotion-0"
            fill="currentColor"
            height="1rem"
            viewBox="0 0 256 256"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16rem"
              width="16rem"
            />
            <line
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
              x1="200"
              x2="56"
              y1="56"
              y2="200"
            />
            <line
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
              x1="200"
              x2="56"
              y1="200"
              y2="56"
            />
          </svg>
           
        </div>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
